### ISSUE TITLE
Check failed: bottom[0]->shape() == bottom[i]->shape()

### ISSUE THREAD

**Issue by grapefruitL** [microsoft/MMdnn#701]
Platform (like ubuntu 16.04/win10):
win10
Python version:
3.6
Source framework with version (like Tensorflow 1.4.1 with GPU):
tf1.12.0 with cpu
Destination framework with version (like CNTK 2.3 with GPU):
caffe
Pre-trained model path (webpath or webdisk path):
https://github.com/grapefruitL/tf2caffe/tree/master/model
Running scripts:
mmconvert -sf tensorflow -in model/model.meta -iw  model/model --dstNodeName Squeeze -df caffe -om outmodel

then i got errs:
```
...
lots of prints
...
I0726 13:33:46.171232 16152 layer_factory.hpp:77] Creating layer conv1_mul_1
I0726 13:33:46.172233 16152 net.cpp:84] Creating Layer conv1_mul_1
I0726 13:33:46.172233 16152 net.cpp:406] conv1_mul_1 <- conv1_sub
I0726 13:33:46.172233 16152 net.cpp:406] conv1_mul_1 <- conv1_mul_1_y
I0726 13:33:46.172233 16152 net.cpp:380] conv1_mul_1 -> conv1_mul_1
F0726 13:33:46.172233 16152 eltwise_layer.cpp:34] Check failed: bottom[0]->shape() == bottom[i]->shape() bottom[0]: 1 10 214 158 (338120), bottom[1]: 1 10 158 214 (338120)
*** Check failure stack trace: ***
```
i think the reason is
https://github.com/microsoft/MMdnn/issues/366#issuecomment-441167240
but i'v no idea how to fix it

**Comment by grapefruitL** [comment]
can i exchange the value of shape[2] and shape[3] of DummyData layer in emit_Constant() function (caffe_emitter.py)? that will solve the problem

another question:
why must the shape[0] of DummyData layer be 1 ?

**Comment by grapefruitL** [comment]
> another question:
> why must the shape[0] of DummyData layer be 1 ?

i found the reason is rename_Placeholder() function in tensorflow_parser.py
`def rename_Placeholder(self, source_node):
        IR_node = self._convert_identity_operation(source_node, new_op='DataInput')
        # shape
        TensorflowParser._copy_shape(source_node, IR_node)
        if len(IR_node.attr['shape'].shape.dim)>0 and len(IR_node.attr['_output_shapes'].list.shape)>0 and len(IR_node.attr['_output_shapes'].list.shape[0].dim)>0:
            IR_node.attr['shape'].shape.dim[0].size = -1
            IR_node.attr['_output_shapes'].list.shape[0].dim[0].size = -1`
what does it mean ?

**Comment by rainLiuplus** [comment]
Hi @grapefruitL, this function implements that add the source_node 'Placeholder' into IR graph and set the first dims of 'shape' and '_output_shape' attributes of IR_node to be -1, which means None.

**Comment by grapefruitL** [comment]
> can i exchange the value of shape[2] and shape[3] of DummyData layer in emit_Constant() function (caffe_emitter.py)? that will solve the problem

seems i'm wrong, when i use the converted caffemodel ,the same err happens, so it means the shape of dummydata generated by the original code is right, why do errors occur during conversion

**Comment by rainLiuplus** [comment]
@grapefruitL, this [line](https://github.com/microsoft/MMdnn/blob/master/mmdnn/conversion/caffe/caffe_emitter.py#L430) of code is wrong. Please change it to `shape = [shape[0], shape[3], shape[1], shape[2]]`

**Comment by grapefruitL** [comment]
@rainLiuplus  thanks! it will solve the problem
